### PR TITLE
Account for super keyword in SpacingAroundParensRule

### LINKED ISSUE #369: spaces is allowed after keyword super
Space after the super keyword is allowed, is this expected?
```
import android.content.Context
import android.util.AttributeSet
import android.view.ContextThemeWrapper
import androidx.appcompat.R.attr.buttonStyle
import androidx.appcompat.widget.AppCompatButton
import com.groupon.maui.components.R

class Test : AppCompatButton {

    @JvmOverloads
    constructor(context: Context, attributeSet: AttributeSet? = null, defStyleAttr: Int = buttonStyle) :
        super (ContextThemeWrapper(context, R.style.ghost_button), attributeSet, defStyleAttr) {
    }
}
```

### PR DESCRIPTION
**Pull request by shashachu**
* When checking for unexpected spaces preceding a `(`, the rule was only checking elementType `IDENTIFIER`, so added a check for `SUPER_KEYWORD`
* Updated lint and formatting tests

### CODE REVIEW

**Comment by shashachu** [comment]
Fixes #369

File: ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/SpacingAroundParensRule.kt [hunk 1]
```diff
@@ -23,7 +29,11 @@ class SpacingAroundParensRule : Rule("paren-spacing") {
             val nextLeaf = node.nextLeaf()
             val spacingBefore = if (node.elementType == LPAR) {
                 prevLeaf is PsiWhiteSpace && !prevLeaf.textContains('\n') &&
-                    prevLeaf.prevLeaf()?.elementType == IDENTIFIER && (
+                    (
```
Comment (shashachu) [inline ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/SpacingAroundParensRule.kt]: This formatting looks pretty wacky to me, but AS seems to agree it's correct.

File: ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/SpacingAroundParensRule.kt [hunk 2]
```diff
@@ -4,13 +4,19 @@ import com.github.shyiko.ktlint.core.Rule
...
 import com.github.shyiko.ktlint.core.ast.ElementType.VALUE_PARAMETER_LIST
 import com.github.shyiko.ktlint.core.ast.nextLeaf
 import com.github.shyiko.ktlint.core.ast.prevLeaf
 import org.jetbrains.kotlin.com.intellij.lang.ASTNode
 import org.jetbrains.kotlin.com.intellij.psi.PsiWhiteSpace
 
+/**
+ * Ensures there are no extra spaces around parentheses.
+ *
+ * See https://kotlinlang.org/docs/reference/coding-conventions.html, "Horizontal Whitespace"
```
Comment (NightlyNexus) [inline ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/SpacingAroundParensRule.kt]: if you want, you can use the fragment: https://kotlinlang.org/docs/reference/coding-conventions.html#horizontal-whitespace

File: ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/SpacingAroundParensRule.kt (hunk 1 above)
Comment (NightlyNexus) [reply ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/SpacingAroundParensRule.kt]: It's weird the this open paren is on a new line but the open paren on line 36 is not.

**Comment by shashachu** [comment]
> Test case?

Tests were added to the lint and format spec files.

File: ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/SpacingAroundParensRule.kt (hunk 2 above)
Comment (shashachu) [reply ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/SpacingAroundParensRule.kt]: ah yes I was looking for that and just looked through it. will update.

File: ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/SpacingAroundParensRule.kt (hunk 1 above)
Comment (shashachu) [reply ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/SpacingAroundParensRule.kt]: Yeah I agree. There's definitely bugs with the indent formatting rule but I can't always figure them out.

**Pull request merged**